<?xml version="1.0"?>

<launch>
  <!-- Radios. -->
   <arg name="uri" default="uri6"/> <!--radio://0/79/2M/" /> -->
  <arg name="uri0" default="radio://0/79/2M/E7E7E7E7E7" />
  <arg name="uri2" default="radio://0/88/2M/E7E7E7E705" />
  <arg name="uri3" default="radio://0/93/2M/E7E7E7E703" />
  <arg name="uri6" default="radio://0/25/2M/E7E7E7E701" />
  <arg name="uri7" default="radio://0/52/2M/E7E7E7E702" />

  <!-- Joystick. -->
  <arg name="joy_dev" default="/dev/input/js0" />

  <!-- Topics. -->
  <arg name="reference_topic" default="reference"/>
  <arg name="state_topic" default="/crazyflie/state"/>
  <arg name="control_topic" default="/crazyflie/control"/>

  <!-- Control parameters. -->
  <arg name="K_file" default="$(find crazyflie_lqr)/references/K_lqr.txt"/>
  <arg name="u_ref_file" default="$(find crazyflie_lqr)/references/u_ref_hover.txt"/>
  <arg name="x_ref_file" default="$(find crazyflie_lqr)/references/x_ref_hover.txt"/>

  <!-- Add a crazyflie. -->
  <include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
    <arg name="uri" value="$(arg $(arg uri))" />
    <arg name="tf_prefix" value="crazyflie" />
    <arg name="enable_logging" value="False" />
  </include>

  <!-- Launch the joystick. -->
  <node name="joy" pkg="joy" type="joy_node" output="screen">
    <param name="dev" value="$(arg joy_dev)" />
  </node>

  <node name="joystick_controller" pkg="crazyflie_demo" type="controller.py" output="screen">
    <param name="use_crazyflie_controller" value="True" />
  </node>

  <!-- Launch the crazyflie controller. -->
  <include file="$(find crazyflie_controller)/launch/crazyflie2.launch">
    <arg name="frame" value="$(arg frame)" />
  </include>

  <!-- Launch the LQR node. -->
  <node name="crazyflie_lqr"
        pkg="crazyflie_lqr"
        type="crazyflie_lqr_node"
        output="screen">

    <!-- Topics. -->
    <param name="reference_topic" value="$(arg reference_topic)"/>
    <param name="state_topic" value="$(arg state_topic)"/>
    <param name="control_topic" value="$(arg control_topic)"/>

    <!-- Control parameters. -->
    <param name="K_file" value="$(arg K_file)"/>
    <param name="u_ref_file" value="$(arg u_ref_file)"/>
    <param name="x_ref_file" value="$(arg x_ref_file)"/>
  </node>
</launch>
